tests/test_synchronize.py

```diff
@@ -21,11 +21,15 @@ def test_semlock_failure():
     from loky.backend.synchronize import SemLock, sem_unlink
 
     name = "loky-test-semlock"
-    sl = SemLock(0, 1, 1, name=name)
-
-    with pytest.raises(FileExistsError):
-        SemLock(0, 1, 1, name=name)
-    sem_unlink(sl.name)
+    try:
+        sl = SemLock(0, 1, 1, name=name)
+
+        with pytest.raises(FileExistsError):
+            SemLock(0, 1, 1, name=name)
+    finally:
+        # Always clean-up the test semaphore to make this test independent of
+        # previous runs (successful or not).
+        sem_unlink(sl.name)
```

Comment:
`sl` is not defined in the case of the failure I suppose:

```pytb
____________________________________________________________ test_semlock_failure _____________________________________________________________
[gw0] darwin -- Python 3.8.12 /Users/glemaitre/mambaforge/envs/dev/bin/python

    @pytest.mark.skipif(sys.platform == "win32", reason="UNIX test")
    def test_semlock_failure():
        from loky.backend.synchronize import SemLock, sem_unlink
    
        name = "loky-test-semlock"
        try:
>           sl = SemLock(0, 1, 1, name=name)

tests/test_synchronize.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <loky.backend.synchronize.SemLock object at 0x1076a09d0>, kind = 0, value = 1, maxvalue = 1, name = 'loky-test-semlock'

    def __init__(self, kind, value, maxvalue, name=None):
        # unlink_now is only used on win32 or when we are using fork.
        unlink_now = False
        if name is None:
            # Try to find an unused name for the SemLock instance.
            for _ in range(100):
                try:
                    self._semlock = _SemLock(
                        kind, value, maxvalue, SemLock._make_name(), unlink_now
                    )
                except FileExistsError:  # pragma: no cover
                    pass
                else:
                    break
            else:  # pragma: no cover
                raise FileExistsError('cannot find name for semaphore')
        else:
>           self._semlock = _SemLock(
                kind, value, maxvalue, name, unlink_now
            )
E           FileExistsError: [Errno 17] File exists

loky/backend/synchronize.py:73: FileExistsError

During handling of the above exception, another exception occurred:

    @pytest.mark.skipif(sys.platform == "win32", reason="UNIX test")
    def test_semlock_failure():
        from loky.backend.synchronize import SemLock, sem_unlink
    
        name = "loky-test-semlock"
        try:
            sl = SemLock(0, 1, 1, name=name)
    
            with pytest.raises(FileExistsError):
                SemLock(0, 1, 1, name=name)
        finally:
            # Always clean-up the test semaphore to make this test independent of
            # previous runs (successful or not).
>           sem_unlink(sl.name)
E           UnboundLocalError: local variable 'sl' referenced before assignment

tests/test_synchronize.py:32: UnboundLocalError
```

Reply:
good catch

Reply:
Should be fixed in the last commit.